src/modules/universal-discovery/components/content-meta-preview/content.meta.preview.component.js

```diff
@@ -10,59 +10,19 @@ export default class ContentMetaPreviewComponent extends Component {
         super();
 
         this.state = {
-            imageUri: null,
+            imageUri: '',
             translations: [],
-            selectContentEnabled: false
+            selectContentEnabled: false,
         };
-    }
-
-    componentDidMount() {
-        this.loadContentInfo(this.props.data.ContentInfo.Content._id);
-    }
-
-    componentWillReceiveProps(props) {
-        if (props.data.id === this.props.data.id) {
-            return;
-        }
-
-        this.setState(state => Object.assign({}, state, {imageUri: null}));
-        this.loadContentInfo(props.data.ContentInfo.Content._id);
-    }
-
-    /**
-     * Loads content info
-     *
-     * @method loadContentInfo
-     * @param {String} contentId
-     * @memberof ContentMetaPreviewComponent
-     */
-    loadContentInfo(contentId) {
-        const promise = new Promise(resolve => this.props.loadContentInfo(this.props.restInfo, contentId, resolve));
 
-        promise
-            .then(this.setImageUri.bind(this))
-            .catch(error => console.log('load:content:info:error', error));
+        this.toggleEnabledState = this.toggleEnabledState.bind(this);
     }
 
-    /**
-     * Sets image URI to image preview
-     *
-     * @method setImageUri
-     * @param {Object} response response from REST endpoint
-     * @memberof ContentMetaPreviewComponent
-     */
-    setImageUri(response) {
-        if (!response.View.Result.count) {
-            return;
-        }
-
-        const version = response.View.Result.searchHits.searchHit[0].value.Content.CurrentVersion.Version;
-        const imageField = version.Fields.field.find(field => field.fieldTypeIdentifier === 'ezimage');
-        const versionLanguages = version.VersionInfo.VersionTranslationInfo.Language;
-        const translations = versionLanguages.map(langauge => this.props.languages.mappings[langauge.languageCode].name);
-        const imageUri = imageField && imageField.fieldValue ? imageField.fieldValue.uri : null;
+    UNSAFE_componentWillReceiveProps(props) {
```

Comment:
As far as I know, we have 16.2 and this is not available in this version? 
Besides that I would use `getDerivedStateFromProps`

Reply:
For some, unknown reason I have React 16.3.2 installed. Everything seems to be working fine. I've also quick checked Alloy Editor and it also works as expected.

Reply:
We should try upgrade in 2.2 to latest React and Bootstrap.

Reply:
We cannot require 16.3 without new tag and tests on assets bundle. 

What we should do:
1. Use old `componentWillReceiveProps`.
2. Upgrade React in assets bundle
3. Test components and React.

Reply:
Yes, let's do it as @dew326 suggested.

Reply:
@lserwatka updating Bootstrap is risky. I remember last time we updated it and it took some time to fix all the issues. If you want to upgrade it, then now is the last moment to do it.

Reply:
@sunpietro that is why we need extensive QA on top of it. Should do it.

Reply:
I fully agree with you both :)